### DISCUSSION TITLE
Add transaction model to ProcessPaymentEvent

### DISCUSSION THREAD

**Discussion by lukeholder** [craftcms/commerce#2450] [Ideas]
Adding the transaction model to the event will allow a developer to dynamically override the transaction’s `paymentType` and set it to either `purchase` or `authorize`. (The default will be set to the gateway’s payment type in config)

https://github.com/craftcms/commerce/blob/develop/src/services/Payments.php#L243

**Comment by rynpsc** [comment]
Has this feature made its way into Commerce yet? We could do with this for a project we are working on where some users payments will be just captured whist others will be authorised as well.
